BuildScript/BuildTasks/TestConfigurationsFactory.cs

```diff
@@ -144,11 +144,46 @@ private string GetPlatform (IEnumerable<string> configurationItems)
 
     private ExecutionRuntime GetExecutionRuntime (IEnumerable<string> configurationItems)
     {
-      var kvp = _supportedExecutionRuntimes.Single (configurationItems, () => CreateMissingConfigurationItemException ("execution runtime"));
+      var enumeratedConfigurationItems = configurationItems.ToArray();
+      if (IsEnforcedLocalMachineAndHasDockerImage (enumeratedConfigurationItems, out var dockerImageOfEnforcedLocalMachine))
+      {
+        return new ExecutionRuntime (MetadataValueConstants.EnforcedLocalMachine, MetadataValueConstants.EnforcedLocalMachine, false, dockerImageOfEnforcedLocalMachine);
+      }
+
+      var kvp = _supportedExecutionRuntimes.Single (enumeratedConfigurationItems, () => CreateMissingConfigurationItemException ("execution runtime"));
 
       var shouldUseDocker = ShouldUseDocker (kvp.Value);
+      var dockerImage = shouldUseDocker ? kvp.Value : "";
+
+      return new ExecutionRuntime (kvp.Key, kvp.Value, shouldUseDocker, dockerImage);
+    }
+
+    private bool IsEnforcedLocalMachineAndHasDockerImage (IEnumerable<string> configurationItems, out string dockerImage)
```

Comment:
Shouldn't this be either `TryGetDockerImage` or `TryGetDockerImageOfEnforcedLocalMachine` to match the 'pattern'?